src/parser/xmlHandler.js

```diff
@@ -118,7 +118,8 @@ class XMLHandler {
   mapObject(node, nsContext, descriptor, val) {
     if (val == null) return node;
     if (typeof val !== 'object' || (val instanceof Date)) {
-      node.text(val);
+      parseXml(node, val, null);
```

Comment:
Fixed. Unintentionally left out parseXML line while trying out correct execution point to convert soapHeader to xml. It's done in base.js for both client and server.